src/api/procedures/pushBenefit.ts

```diff
@@ -0,0 +1,118 @@
+import { DividendDistribution, Identity, PolymeshError, Procedure } from '~/internal';
+import { ErrorCode, RoleType, TxTags } from '~/types';
+import { ProcedureAuthorization } from '~/types/internal';
+import { tuple } from '~/types/utils';
+import {
+  boolToBoolean,
+  corporateActionIdentifierToCaId,
+  identityIdToString,
+  stringToIdentityId,
+} from '~/utils/conversion';
+
+export interface PushBenefitParams {
+  targets: (string | Identity)[];
+}
+
+/**
+ * @hidden
+ */
+export type Params = { distribution: DividendDistribution } & PushBenefitParams;
+
+/**
+ * @hidden
+ */
+export async function preparePushBenefit(
+  this: Procedure<Params, void>,
+  args: Params
+): Promise<void> {
+  const {
+    context: {
+      polymeshApi: { tx, query },
+    },
+    context,
+  } = this;
+  const {
+    distribution: { id: localId, ticker, paymentDate, expiryDate },
+    targets,
+  } = args;
+
+  const now = new Date();
+
+  if (paymentDate > now) {
+    throw new PolymeshError({
+      code: ErrorCode.ValidationError,
+      message: "The Distribution's benefits date hasn't been reached",
+      data: { paymentDate },
+    });
+  }
+
+  if (expiryDate && expiryDate < now) {
+    throw new PolymeshError({
+      code: ErrorCode.ValidationError,
+      message: 'The Distribution has already expired',
+      data: {
+        expiryDate,
+      },
+    });
+  }
```

Comment:
These same validations are being made in the `claimDividends` procedure. They should probably be refactored into an `assertDistributionOpen` util